src/lib/form-field/form-field.ts

```diff
@@ -344,10 +344,17 @@ export class MatFormField extends _MatFormFieldMixinBase
     return this._hasLabel() || this.appearance === 'legacy' && this._hasPlaceholder();
   }
 
-  /** Determines whether to display hints or errors. */
-  _getDisplayedMessages(): 'error' | 'hint' {
-    return (this._errorChildren && this._errorChildren.length > 0 &&
-        this._control.errorState) ? 'error' : 'hint';
+  /** Determines whether to display hints, errors or none. */
+  _getDisplayedMessages(): 'error' | 'hint' | 'none' {
+    return this.areErrorMessagesVisible() ? 'error' : this.areHintsVisible() ? 'hint' : 'none';
+  }
+
+  private areErrorMessagesVisible(): boolean {
```

Comment:
This method is missing docs. Also we prefix private method names with an underscore.